### DISCUSSION TITLE
BaseSClassifier with catboost for multiple treatment

### DISCUSSION THREAD

**Discussion by omri-al** [uber/causalml#746] [Q&A]
Hi,
I'm sorry for the non-specific issue here.
I want to use causalML package for the use case of multiple treatment (binary response). 
My data consists mostly of categorical features, so I have chose to use catboost as the base model.

The issue is, no matter what I tried, I couldn't make it work (training BaseSClassifier, with catboost as bas model, with categorical features).
When I try to separately train the catboost model it works without any issue.
however, using the BaseSClassifier, I get issues in the train or predict stages.

couldn't find a specific example for this in the internet.
Have anyone faced similar issue?
Thanks

**Comment by jeongyoonlee** [comment]
I've tested `BaseTClassifier` with `CatBoostClassifer` as a base learner and it worked. You can find the colab notebook [here](https://colab.research.google.com/drive/1yyUX8Qdv-0yIIPyKSfpibZSJDwCzVCZQ?usp=sharing).

If you keep having issues, please share your code so I can understand the issue better.

Hope it helps.

**Reply by levanjon** [reply]
there seems to be an issue when you try to pass cat_features to catboost - it does not work for them.

**Comment by DSMaster534** [comment]
When using CatBoost in a causal inference context with the BaseSClassifier from the CausalML library, it is important to understand how CatBoost handles input data transformation.

CatBoost transforms the input data  into a format optimized for its internal processes. This transformation often involves creating new features derived from the original variables, which are then used for training and predictions. Once this transformation is complete, the data becomes read-only, meaning the values of the input variables, including the treatment variable, cannot be modified.

Within the BaseSClassifier, this behavior creates a fundamental incompatibility with CatBoost. The BaseSClassifier expects  to remain separate from the features in  throughout the model training process, allowing the algorithm to handle  independently. However, CatBoost's transformation process does not allow for such separation, as any variable included in  is transformed and becomes immutable.

As a result, BaseSClassifier cannot be used with CatBoost due to these differences in data handling requirements. If CatBoost is a desired model, alternative implementations or workarounds that respect the causal inference framework must be considered.
